lib/portage/dbapi/bintree.py

```diff
@@ -1561,6 +1703,9 @@ def getname(self, cpv, allocate_new=None):
 		except AttributeError:
 			cpv = _pkg_str(cpv)
 
+		if binpkg_format is None:
+			binpkg_format = self.settings.get("BINPKG_FORMAT", "xpak")
```

Comment:
The `getname` method is similar to the `_pkgindex_entry` method in the sense that we really should not assume a constant binpkg_format here.